### PR TITLE
feat(type): add  isSafeInteger

### LINKED ISSUE #747: isSafeInteger
Determines whether the provided value is a number that is a safe integer.

Complete description of what safe integer is, can be found here on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger). The linked page also contains polyfill and preferred implementation of the `isSafeInteger` function. Modern javascript already contains function `Number.isSafeInteger`.

Expected behavior
```js
RA.isSafeInteger(3);                    // true
RA.isSafeInteger(Math.pow(2, 53));      // false
RA.isSafeInteger(Math.pow(2, 53) - 1);  // true
RA.isSafeInteger(NaN);                  // false
RA.isSafeInteger(Infinity);             // false
RA.isSafeInteger('3');                  // false
RA.isSafeInteger(3.1);                  // false
RA.isSafeInteger(3.0);                  // true
```

[This PR](https://github.com/char0n/ramda-adjunct/pull/1067) should serve as a model template of PR for `isSafeInteger` function.

### PR DESCRIPTION
**Pull request by rostyslav-diakiv**
Closes #747

### CODE REVIEW

**Comment by rostyslav-diakiv** [comment]
@char0n 
Hi, I finished working on the feature, can you take a look, please =)

**Comment by rostyslav-diakiv** [comment]
> I realized after the review, we also need to test the pollyfil. How to do that can be found e.g. in this PR: We also need to test the pollyfil. How to do that can be found in this file:
> 
> https://github.com/char0n/ramda-adjunct/blob/7f70813b73f41275ccca6b8983d4b9bb235f5d44/test/toArray.js#L77
> 
> The point is to explicitly test the public api function but also the internal polyfill. That way we guarantee that the nonpolyfilled native function and polyfill behaves the same.

Sure, I am on it)

**Comment by rostyslav-diakiv** [comment]
Hello again, I've made a chenges you requested, can you take a look?)

File: test/isSafeInteger.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        assert.isFalse(isSafeIntegerPolyfill(3.1));
+        assert.isFalse(isSafeIntegerPolyfill('string'));
+        assert.isFalse(isSafeIntegerPolyfill(null));
+        assert.isFalse(isSafeIntegerPolyfill(undefined));
+        assert.isFalse(isSafeIntegerPolyfill({}));
+        assert.isFalse(isSafeIntegerPolyfill(() => {}));
+        assert.isFalse(isSafeIntegerPolyfill(true));
+      });
+    });
+  });
```
Comment (char0n) [inline test/isSafeInteger.js]: We're missing test for polyfill supporting the currying. The same as this one: https://github.com/char0n/ramda-adjunct/pull/1112/files#diff-bdc4a9deeaf42a6439119d62632e80ffR33 but for polyfill.

Comment (rostyslav-diakiv) [reply test/isSafeInteger.js]: Added test for polyfill supporting the currying

File: test/isSafeInteger.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  specify('should support placeholder to specify "gaps"', function() {
+    const isSafeInteger = RA.isSafeInteger(R.__);
+
+    assert.isTrue(isSafeInteger(3));
+  });
+
+  context('fromPolyfill', function() {
```
Comment (char0n) [inline test/isSafeInteger.js]: `fromPolyfill -> isSafeIntegerPolyfill`. I'll fix this myself after the merge.

Comment (char0n) [reply test/isSafeInteger.js]: Fixed in 0a7568c2040bb8111a7a1336b8342a86f895c948

**Pull request merged**